### DISCUSSION TITLE
Issues using script

### DISCUSSION THREAD

**Discussion by sag2643** [netbox-community/netbox#13719] [General]
Hi all

The ips are reachable (pingable) from netbox server, when i run script (netbox-ipscanner),all the IPs which are reachable from server are markes as  "deprecated" and if i try to run the script couple of times then the status marks as active . The status keeps on changing either to deprecated or active

In the log i see it as "Host 192.168.90.3/24 exists in netbox but not responding --> DEPRECATED" -- but this ip is reachable from netbox . Any suggestions why its not scanning properly. Is there a way to set the rate limit on the netbox api or to the scanner to set?

**Comment by candlerb** [comment]
This script "netbox-ipscanner" is not part of Netbox.

Therefore, if you still have a problem with the functioning of this third-party script, please take it up with the author of that script - or else modify it yourself to make it work the way you want it to work (the latter option will involve learning Python coding)

Please also note this paragraph from the [Netbox documentation](https://docs.netbox.dev/en/stable/introduction/#serve-as-a-source-of-truth):

> NetBox intends to represent the _desired_ state of a network versus its _operational_ state. As such, automated import of live network state is strongly discouraged. All data created in NetBox should first be vetted by a human to ensure its integrity. NetBox can then be used to populate monitoring and provisioning systems with a high degree of confidence.

**Reply by sag2643** [reply]
is there any limit with the api adding ipaddresses to the netbox?

**Reply by candlerb** [reply]
What sort of limit? No, the API can manage IP addresses the same as you can do via the web interface. There are no rate limits or similar.

In any case, the problem you are describing is with the script itself (being unable to detect accurately whether a host is up or down). You will need to debug the script.

**Reply by sag2643** [reply]
If the script has issues, then when i execute the script and tcpdump simultaneously, i can see the active ipaddresses on the netbox server are listed correctly. but it just doesnot get added.

**Reply by candlerb** [reply]
Your script (not Netbox) is logging the message you reported:

> Host 192.168.90.3/24 exists in netbox but not responding --> DEPRECATED

Clearly then, it is probing the host, detecting it as down, then telling Netbox to update its status to DEPRECATED. Furthermore, we know that the update is happening successfully, as you reported that the status does indeed change to DEPRECATED in Netbox.

Netbox is performing perfectly. Your script is broken.

I am exiting this conversation now.
